### PR TITLE
Removed the functions from the loop as it as executing n times.

### PR DESCRIPTION
**Pull request by myntuser**
Removed the functions in the loop. count() was taken out of the loop.

### CODE REVIEW

**Comment by domnulnopcea** [comment]
:+1:

File: src/Ubirimi/Documentador/Repository/Space/Space.php [hunk 1]
```diff
@@ -423,7 +423,7 @@ public function updatePermissionsAnonymous($spaceId, $parameters) {
         $query = 'insert into documentator_space_permission_anonymous(documentator_space_id, ';
         $parametersCount = count($parameters);
 
-        for ($i = 0; $i < count($parameters); $i++) {
+        for ($i = 0; $i < $parametersCount; $i++) {
```
Comment (domnuprofesor) [inline src/Ubirimi/Documentador/Repository/Space/Space.php]: defining an extra $count variable does not help making the code clear unless you use it later somewhere else. This usually is not the case. So I propose to write the `for` as:
`for ($i = 0, $count = count($parameters); $i < $count; $i++)

**Comment by domnulnopcea** [comment]
:+1:

File: src/Ubirimi/Documentador/Repository/Space/Space.php (hunk 1 above)
Comment (w2pc) [inline src/Ubirimi/Documentador/Repository/Space/Space.php]: IMO it makes the code cleaner but the main reason for moving count() to a predefined line is that it keeps count() from being called on every iteration of the loop
for ($i = 0, $count = count($parameters); $i < $count; $i++)
will still cause count() to be called n times

Comment (domnuprofesor) [inline src/Ubirimi/Documentador/Repository/Space/Space.php]: IMHO, the first part of the for is the initialization part that gets called only once; the second and third part gets called n times

Francisc Nopcea
founder
History Lapse

+40 752 170 174

Comment (w2pc) [inline src/Ubirimi/Documentador/Repository/Space/Space.php]: Yes, you are correct. I miss read the way you wrote the new for() count()
would only get called once.

**Pull request merged**